File: Notes/codes/c/01_simple.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv)
{
  int ierr, Nprocs, my_rank;

  ierr = MPI_Init(&argc, &argv);
  ierr = MPI_Comm_size(MPI_COMM_WORLD, &Nprocs);
  ierr = MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
  printf("Hello, I am proc: %d from %d processes\n", my_rank, Nprocs);
  ierr = MPI_Finalize();
  
  return 0;
}

File: Notes/codes/f90/01_simple.f90
PROGRAM simple
  IMPLICIT NONE
  INCLUDE 'mpif.h'
  INTEGER :: my_rank, Nprocs, ierr
  CALL MPI_Init(ierr)
  CALL MPI_Comm_size(MPI_COMM_WORLD, Nprocs, ierr)
  CALL MPI_Comm_rank(MPI_COMM_WORLD, my_rank, ierr)
  WRITE(*,*) 'Hello, I am proc: ', my_rank, ' from ', Nprocs, ' total processes'
  CALL MPI_Finalize(ierr)
END PROGRAM

